### ISSUE TITLE
Sort By… (Name Recently added…)

### ISSUE THREAD

**Issue by 919Studios** [alefragnani/vscode-bookmarks#217]
Check out the screenshot.  I was wondering if you would be able to add a sorting option like this to bookmarks settings:
![can_this_be_added_to_bookmarks_feature_request](https://user-images.githubusercontent.com/20955174/59317215-ef07f980-8c8f-11e9-9c42-48c310f718cb.png)
Bookmarks is really helpful.  Thank you for all your work on it!

**Comment by alefragnani** [comment]
Hi @919Studios,

First of all, thank you for your kind words. I'm glad to know you like the extension! 👍 

Right now, the bookmarks are sorted only by `Line Number` and grouped by `File` and `Workspace`. Support sorting by `Name`(label / line preview) and `Path` should be straightforward (grouped or not), but `Recent`, on the other hand, is not. There is no such info in its engine, and I'm not sure the easiest way to do that, but I'll take a look.

Thank you for your suggestion.

**Comment by fabb** [comment]
Being used to the Bookmark feature of IntelliJ, I really miss bookmarks being sorted by "Recent" by default (most recently added on the top).  Always when I open the bookmark list, I am very confused on where the bookmark went I just added, and I need to go through all of them one by one, which defeats the purpose of the bookmarks extension saving time. This feature would improve this great extension a lot.

**Comment by alefragnani** [comment]
Hi @fabb ,

I never used IntelliJ itself, just Rider recently, but I guess both share the same Bookmarks functionality. Honestly I don't remember how it works, but I'll try to revisit it. 

Personally, I don't miss any sorting at all, because in my workflow, when I'm _finding_ bookmarks, I prefere to _type_ the content that I'm looking for (instead of visually look for the bookmarks), filtering the results. But I totally understand the need 👍.

To me, the _Recent_ sorting is more like a _Stacked Bookmark_, present in some other editors. Just like I commented above, it needs reengineering ..

Hope this helps

**Comment by fabb** [comment]
Yeah, should be the same in Rider, it‘s also the same in AppCode.
For me it‘s like a clipboard history which is also sorted chronologically, fits like how I work with it.

**Comment by liushichuan** [comment]
Hi, bookmarks are great and very useful, I use them to help read c++ source code, if the tags can be sorted according to the time of creation, it will be a great tool for sorting out the code flow

**Comment by B0J1** [comment]
Another extension https://github.com/alefragnani/vscode-numbered-bookmarks is what I want. But how about merging it to this extension?

**Comment by alefragnani** [comment]
HI @B0J1 , 

No intention. Both extensions shares some features (that makes sense), but have two different objectives. 

The Numbered Bookmarks extension was created to mimic the Delphi experience I loved throughout the years, and I'm keeping it simpler. This one, on the other hand, is a more featured experience with Side Bar, Selection, Customizations, Tags, etc. And plans to evolve it even more.

**Comment by DeathPoem** [comment]
what if some provide this feature and make a merge request ?

**Comment by DeathPoem** [comment]
what if someone provide this feature and make a merge request ?

**Comment by yurymann** [comment]
Just wanted to bump this request as it seems to be stale. I'm another former IntelliJ user who is missing the most-recently-used sorting. Even though this is the best vscode bookmarks extension I've found, I almost stopped using bookmarks because it's often quicker to find the file or symbol again than visually scan the Bookmarks view. In IntelliJ, I used bookmarks all the time: it's very efficient to open the Bookmarks window and select between last 2 or 3 bookmarks with a couple of key presses, without deleting the other 10 bookmarks which may still be useful.

**Comment by z1yuan** [comment]
https://github.com/alefragnani/vscode-bookmarks/pull/851
i made a PR to solve this
